### ISSUE TITLE
When is VoxCPM2 getting released?

### ISSUE THREAD

**Issue by tensorjackal** [OpenBMB/VoxCPM#192]
I saw that @a710128 have made changes in his nanovllm implementation for adding VoxCPM2 support. When will it be released? What updates will it have?

**Comment by liuxin99** [comment]
Thanks for your interest in VoxCPM2! We are still testing it internally and don’t have a specific release date to share yet, but please stay tuned for upcoming updates.

**Comment by tensorjackal** [comment]
Amazing! Excited for what new features/languages would it bring.

**Comment by alkemiik-coder** [comment]
Hey guys, great job with voxCPM — seriously, huge respect.

I completed a full SFT run for Polish using a dataset of 10,000 hours of audio on an RTX 5090. The training took 45 hours, and the model performs incredibly well. Naturally, data preparation was a challenge, but I managed to get it done.

Are you planning to fully expose and support parameters such as SEED and TEMPERATURE?

For inference, I’m using a fork of nano-vLLM, and it is insanely fast. On an RTX 5090, with VRAM fully utilized and using STEPS=40 / CFG=1.9, I can generate one hour of audio in just 11 minutes. That’s an excellent result — in theory, I’m able to generate an entire audiobook in about one hour.

I’m planning to upgrade to an RTX 6000 Pro / DGX Spark, and I’d love to ask for your engineering recommendations.

Personally, I have to say this is the only TTS engine that, after extensive testing (xttsv2, chatterbox, tada, fish), does not produce hallucinations.

I’m currently preparing a larger training dataset, and I’d like to know which settings your engine prefers for the most optimal training results.

I really can’t wait for the next model.

Thanks a lot!

**Comment by madey83** [comment]
> Hey guys, great job with voxCPM — seriously, huge respect.
> 
> I completed a full SFT run for Polish using a dataset of 10,000 hours of audio on an RTX 5090. The training took 45 hours, and the model performs incredibly well. Naturally, data preparation was a challenge, but I managed to get it done.
> 
> Are you planning to fully expose and support parameters such as SEED and TEMPERATURE?
> 
> For inference, I’m using a fork of nano-vLLM, and it is insanely fast. On an RTX 5090, with VRAM fully utilized and using STEPS=40 / CFG=1.9, I can generate one hour of audio in just 11 minutes. That’s an excellent result — in theory, I’m able to generate an entire audiobook in about one hour.
> 
> I’m planning to upgrade to an RTX 6000 Pro / DGX Spark, and I’d love to ask for your engineering recommendations.
> 
> Personally, I have to say this is the only TTS engine that, after extensive testing (xttsv2, chatterbox, tada, fish), does not produce hallucinations.
> 
> I’m currently preparing a larger training dataset, and I’d like to know which settings your engine prefers for the most optimal training results.
> 
> I really can’t wait for the next model.
> 
> Thanks a lot!

Hey!

Really interesting post — I've been looking into Polish TTS myself better than xttsv2 currently used. You mentioned that results are excellent, does it mean it is similar quality like eleven labs or better?

Any chance you'd release the Polish model ?And would you mind sharing where you sourced 10,000 hours of Polish audio and how you prepared the dataset?

Thanks!

**Comment by vytskalt** [comment]
Honestly this is the best open-source TTS I've seen so far. The voice cloning is amazing, and after fine-tuning on 24 hours of a specific voice I can get output virtually indistinguishible from the real human. Big props to the team. The only issue I came across was the voice tending to slowly drift in pitch over the output, making seamless concatenation of different chunks harder. Excited to see what comes next.

**Comment by alkemiik-coder** [comment]
@madey83 

The source of my dataset is Mozilla Common Voice / wolnelektury.pl / projekt-gutenberg.org. I processed the audio using RipX and demucs_4 to isolate only the spoken parts. Then I built a program that uses VAD_silero to split the input audio into segments of the desired length, while making sure they are not cut in the middle of a word. After that, I developed an application that uses whisperv3 to transcribe those files and generate a JSON file with the transcription and the appropriate structure. Naturally, the data still has to be checked manually to make sure it has been processed correctly, and only then can the training scripts be used.

Up to this point, I had been using xttsv2, but then I switched to chatterbox. However, both engines hallucinate, and to trim those hallucinations I used whisperx and Montreal Forced Aligner. Still, these are not ideal solutions, since they either slow the workflow down or trim incorrectly, or too aggressively. With voxCPM, I do not have those problems, although the silence generated at the beginning of the audio is too long, so I use post-processing scripts that automatically trim that silence.

**Comment by madey83** [comment]
> [@madey83](https://github.com/madey83)
> 
> The source of my dataset is Mozilla Common Voice / wolnelektury.pl / projekt-gutenberg.org. I processed the audio using RipX and demucs_4 to isolate only the spoken parts. Then I built a program that uses VAD_silero to split the input audio into segments of the desired length, while making sure they are not cut in the middle of a word. After that, I developed an application that uses whisperv3 to transcribe those files and generate a JSON file with the transcription and the appropriate structure. Naturally, the data still has to be checked manually to make sure it has been processed correctly, and only then can the training scripts be used.
> 
> Up to this point, I had been using xttsv2, but then I switched to chatterbox. However, both engines hallucinate, and to trim those hallucinations I used whisperx and Montreal Forced Aligner. Still, these are not ideal solutions, since they either slow the workflow down or trim incorrectly, or too aggressively. With voxCPM, I do not have those problems, although the silence generated at the beginning of the audio is too long, so I use post-processing scripts that automatically trim that silence.

Hi 

Thanks for replying.

I I also use some trim cutting solution for xttsv2 hallucinations... It's fine and fast.

I have tried to fine-tuning xttsv2 and prepared dataset (6h audio) using this https://github.com/JarodMica/dataset-maker plus uvr5 with Kim vocal 2.

And in addition custom script to change digits to words with whisperx and gpt.

Fine-tuning results were very badly due to fact that new model produces even more hallucinations than original xttsv2.

Now I'm looking better tts model which will produce more clear voice and it is having better cloning capabilities.

I don't have RTX 5090. I use RTX 2060 for xttsv2.

**Comment by madey83** [comment]
@alkemiik-coder 

Hey, thanks for the detailed response!

A few questions:

1. What is the minimum amount of hours you think would still produce acceptable Polish quality — did you experiment with smaller datasets before going to 10,000h?

2. How many different speakers are in your dataset? Is speaker diversity important for language learning, or did you use mostly single-speaker audio?

3. Could you share any audio samples of the Polish output? It would really help to hear the quality before investing time in dataset preparation.

Thanks!

**Comment by alkemiik-coder** [comment]
@madey83 
Unfortunately, I still can’t share the model yet because it is still too weak. It makes mistakes on very specific words, and I am currently working on synthetic voices that will better match the model. That is also why I asked the voxCPM team the questions above, specifically about the parameters for SFT training when adding another language (or introducing new tokens in the tokenizer to represent new languages).

Maybe the voxCPM team will surprise us and release a 3B model. Such a model would have many more parameters for learning and working on pronunciation, and it would likely handle training on other datasets much better.

Personally, I do not know what the minimum is. I experimented with a 100-hour dataset and 10 hours of training, as well as a 1,000-hour dataset and 15 hours of training, and the results were very similar to this training run with a 10,000-hour dataset and 45 hours of training. So, most likely, my technical parameters were not chosen optimally. That is why I am asking the architects which parameter ranges make sense for which dataset sizes. I used 25 different speakers, defining them appropriately in the JSON transcript through ref_audio.

As for my current samples, here you go. They are attached below.

[01.mp3](https://github.com/user-attachments/files/26112745/01.mp3) | [02.mp3](https://github.com/user-attachments/files/26112743/02.mp3) | [03.mp3](https://github.com/user-attachments/files/26112747/03.mp3) | [04.mp3](https://github.com/user-attachments/files/26112744/04.mp3) | [Source text](https://github.com/user-attachments/files/26112746/text.txt)

I generated the audio on an RTX 3060.

**Comment by madey83** [comment]
Thanks

**Comment by progmars** [comment]
> [@madey83](https://github.com/madey83) 
> Personally, I do not know what the minimum is. I experimented with a 100-hour dataset and 10 hours of training, as well as a 1,000-hour dataset and 15 hours of training, and the results were very similar to this training run with a 10,000-hour dataset and 45 hours of training. 

Thanks for sharing your experience. I used similar approach for Latvian, using just about 20h of data from Mozilla Common Voice. Not much Latvian data available especially after cleaning out their Unicode-messed entries that had "?" chars for special symbols. 

The quality of Mozilla Common Voice samples is, well, random :D People recording with cheap mics in noisy environments. Still, after just a few days of training, VoxCPM could talk smooth Latvian with just occasional mispronunciations. It helps that Latvian language is 99.99% "read what you see" language with exact spelling/writing correlation. The only problem is letter o which is pronounced as "ua" in native Latvian words and as short "o" in international words. I'm almost thinking of adding a vocabulary to improve this, but then I'll need to collect all internationalisms with "o", which is too much work. Also, working on a better dataset, but it's tricky. We have entire public radio recordings archive, but they haven't given explicit permission to use the data for machine learning, so I guess, I could use it only for my completely private model.

And yeah, the voice drifting can be an issue. With time, the voice tends to get lower, a bit raspy and metallic. Tweaking CFG helps a bit, but still. I hope that VoxCPM2 would make it more consistent. And, of course, a better dataset might help as well.

**Comment by alkemiik-coder** [comment]
Information about voxCPM2 was added today, and I’m as excited as ClaudeOpus about the new GPUs from nVidia. Maybe they’ll introduce one multimodal model and another Base model with an improved architecture — that would be a real treat. I’d gladly take it to the workshop for some hobby testing. It would also be great if they added a clear table with recommendations for training new languages or dataset sizes, along with what one can potentially expect from such training.
